sway/tree/container.c

```diff
@@ -23,6 +25,30 @@ void swayc_descendants_of_type(swayc_t *root, enum swayc_types type,
 	}
 }
 
+static void update_root_geometry() {
```

Comment:
You can get output layout extents with `wlr_output_layout_get_box(layout, NULL)`.